node/avn-service/src/ethereum_events_handler.rs

```diff
@@ -349,27 +349,29 @@ where
 
     log::info!("⛓️  ETH EVENT HANDLER INITIALIZED");
 
-    let public_keys = config.keystore.sr25519_public_keys(AVN_KEY_ID);
-
-    let author_public_keys = config
-        .client
-        .runtime_api()
-        .query_author_signing_keys(config.client.info().best_hash)
-        .map_err(|e| {
-            log::error!("Error querying authors: {:?}", e);
-        })
-        .and_then(|opt_keys| match opt_keys {
-            Some(keys) => Ok(keys),
-            None => Err(()),
-        });
-
-    let current_node_public_key = match find_author_account_id(author_public_keys, public_keys) {
-        Some(key) => key,
-        None => {
-            log::error!("Author not found");
-            return
-        },
-    };
+    let mut current_node_public_key = Public([0u8; 32]);
+    loop {
```

Comment:
Am not sure if we want to loop indefinitly. Perhaps a set number of retries should be done before we give us and resume operations without throwing an error and killing the node?

Reply:
If its not busy waiting, and this is an essential task for the operation of a collator, why shouldn't we loop till we achieve the task?